Modules/timemodule.c

```diff
@@ -2122,36 +2171,41 @@ pysleep(_PyTime_t secs)
             Py_BEGIN_ALLOW_THREADS
             Sleep(ul_millis);
             Py_END_ALLOW_THREADS
+            ret = 0;
             break;
         }
 
-        hInterruptEvent = _PyOS_SigintEvent();
-        ResetEvent(hInterruptEvent);
+        hEvents[EVENT_INTERRUPT] = _PyOS_SigintEvent();
+        ResetEvent(hEvents[EVENT_INTERRUPT]);
 
         Py_BEGIN_ALLOW_THREADS
-        rc = WaitForSingleObjectEx(hInterruptEvent, ul_millis, FALSE);
+        rc = WaitForMultipleObjectsEx(EVENT_SIZE, hEvents, FALSE, ul_millis, FALSE);
         Py_END_ALLOW_THREADS
 
-        if (rc != WAIT_OBJECT_0)
+        if (rc == WAIT_FAILED) {
+            ret = -1;
             break;
-#endif
+        }
 
-        /* sleep was interrupted by SIGINT */
-        if (PyErr_CheckSignals()) {
-            return -1;
+        if ((rc == (WAIT_OBJECT_0 + EVENT_TIMER)) || (rc == WAIT_TIMEOUT)) {
+            ret = 0;
+            break;
         }
 
-#ifndef HAVE_CLOCK_NANOSLEEP
-        if (get_monotonic(&monotonic) < 0) {
-            return -1;
+        if (rc != (WAIT_OBJECT_0 + EVENT_INTERRUPT)) {
+            ret = 0;
+            break;
         }
-        secs = deadline - monotonic;
-        if (secs < 0) {
+
+        /* sleep was interrupted by SIGINT */
+        if (PyErr_CheckSignals()) {
+            ret = -1;
             break;
         }
-        /* retry with the recomputed delay */
```

Comment:
Please add a comment since this code path is non-obvious.

```
// The SIGINT signal handler was called and did not raise a Python exception.
// Restart sleeping. The timer deadline doesn't need to be recomputed:
// SetWaitableTimer() uses absolute time.
```

Reply:
Done